File: templates/ruflet_flutter_template/apple_packages/ruflet_apple/Sources/RufletExtensions/RufletPermissionHandler/Sources/Extension.swift
import RufletEngine

@MainActor
public struct RufletPermissionHandlerExtension: RufletExtension {
  public init() {}

  public func createService(for control: RufletControl) -> RufletService? {
    control.type == "PermissionHandler" ? PermissionHandlerService(control: control) : nil
  }
}

File: templates/ruflet_flutter_template/ruflet_packages/ruflet_permission_handler/lib/src/extension.dart
import 'package:ruflet/ruflet.dart';

import 'permission_handler.dart';

class Extension extends RufletExtension {
  @override
  RufletService? createService(Control control) {
    switch (control.type) {
      case "PermissionHandler":
        return PermissionHandlerService(control: control);
      default:
        return null;
    }
  }
}
